### ISSUE TITLE
Issue with simple Gaussian distribution

### ISSUE THREAD

**Issue by v-i-s-h** [nonconvexopt/pytorch-SSGE#2]
Hi,

First of all thanks for this wonderful implementation.

I am now trying to work a simple example of using SSGE for estimating score function for N(0, 1) function. Please find below my code:
```python
import torch
import gpytorch
from ssge import SSGE

def test_gaussian():
    import matplotlib.pyplot as plt
    plt.style.use('seaborn')

    dist = torch.distributions.normal.Normal(
        torch.tensor(0.0), torch.tensor(1.0)
    )

    kernel = gpytorch.kernels.RBFKernel()

    # estimator = SSGE(gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel()))
    estimator = SSGE(
        gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel()), 
        noise=1e-4
    )

    sample = dist.sample((100, 1))
    logp = dist.log_prob(sample)
    sample.requires_grad_(True)
    estimator.fit(sample)


    # test_points = dist.sample((25, 1))
    test_points = torch.linspace(-2, +2, 250)
    test_points.requires_grad_(True)
    grad_the = -test_points
    grad_est = estimator(test_points)

    # Plot results
    fig, ax = plt.subplots(1, 1)
    ax.scatter(sample.detach().numpy(), logp.detach().numpy(),
               marker='o', s=2.0, label='Sample points')
    ax.plot(test_points.detach().numpy(), grad_the.detach().numpy(),
            c='red', lw=1, label=r"$\nabla_x")
    ax.plot(test_points.detach().numpy(), grad_est.detach().numpy(),
            c='blue', lw=1,  label=r"$\hat{\nabla_x}")
    plt.savefig("out/test_gaussian.png")

test_gaussian()
```

However, when I run the example, the estimated gradients are very much different than the actual ones. (See the figure attached)

How this can be improved?

![test_gaussian](https://user-images.githubusercontent.com/7954412/148080955-ddfde33c-658e-4223-852f-f5175aff9d6f.png)

**Comment by nonconvexopt** [comment]
Thank you for your interest to this repo. I am sorry for the inconvinience. I will work on this issue in this week.

**Comment by nonconvexopt** [comment]
I assume the problem is at the core algorithm. I will check it again soon.

**Comment by nonconvexopt** [comment]
Sorry for the late response. In the test you generated, average value of the estimated gradient (blue line) seems to resemble the log density, not the derivative of the log density. It is highly probable that I implemented something wrong and I am analyzing the original paper again.

However, I guess the noise in gradient estimate are expected even in the complete implementation, because of 2 reasons:
1) Optimization with respect to Kernel Hyperparameter is needed. In the simulated example like analytical distributions, for optimization, loss can be just MSE of estimated gradient and theoretical gradient. I don't understand what is the loss for the implicit distribution for now.
2) Small sample size.
